include/kos/init.h

```diff
@@ -85,14 +108,24 @@ extern void * __kos_romdisk;
 #define INIT_DEFAULT \
     (INIT_IRQ | INIT_THD_PREEMPT)
 
-#define INIT_NONE           0x0000  /**< \brief Don't init optional things */
-#define INIT_IRQ            0x0001  /**< \brief Enable IRQs at startup */
+#define INIT_NONE           0x00000000  /**< \brief Don't init optional things */
+#define INIT_IRQ            0x00000001  /**< \brief Enable IRQs at startup */
 /* Preemptive mode is the only mode now. Keeping define for compatability. */
-#define INIT_THD_PREEMPT    0x0002  /**< \brief Enable thread preemption */
-#define INIT_NET            0x0004  /**< \brief Enable built-in networking */
-#define INIT_MALLOCSTATS    0x0008  /**< \brief Enable malloc statistics */
-#define INIT_QUIET          0x0010  /**< \brief Disable dbgio */
-#define INIT_EXPORT         0x0020  /**< \brief Export kernel symbols */
+#define INIT_THD_PREEMPT    0x00000002  /**< \brief Enable thread preemption */
+#define INIT_NET            0x00000004  /**< \brief Enable built-in networking */
+#define INIT_MALLOCSTATS    0x00000008  /**< \brief Enable malloc statistics */
+#define INIT_QUIET          0x00000010  /**< \brief Disable dbgio */
+#define INIT_EXPORT         0x00000020  /**< \brief Export kernel symbols */
+#define INIT_CONTROLLER     0x00000040  /**< \brief Enable Controller maple driver */
+#define INIT_KEYBOARD       0x00000080  /**< \brief Enable Keyboard maple driver */
+#define INIT_MOUSE          0x00000100  /**< \brief Enable Mouse maple driver */
+#define INIT_LIGHTGUN       0x00000200  /**< \brief Enable Lightgun maple driver */
+#define INIT_VMU            0x00000400  /**< \brief Enable VMU maple driver */
+#define INIT_PURUPURU       0x00000800  /**< \brief Enable Puru Puru maple driver */
+#define INIT_SIP            0x00001000  /**< \brief Enable Sound input maple driver */
+#define INIT_DREAMEYE       0x00002000  /**< \brief Enable DreamEye maple driver */
```

Comment:
These definitely should not be in the arch-independent file. They should be in kernel/arch/dreamcast/include/arch/init_flags.h

Reply:
Oh yeah, good call!